src/widgets/PrivacyWidget.php

```diff
@@ -0,0 +1,165 @@
+    public $privacyMessage;
+
+    /**
+     * @var string Link to the privacy policy page
+     */
+    public $messageLink = '';
+    
+    /**
+     * @var string Text on the accept button
+     */
+    public $acceptPrivacyButtonText;
+
+    /**
+     * @var string Accept privacy policy button css class
+     */
+    public $acceptPrivacyButtonClass = 'btn btn-primary';
+
+    /**
+     * @var bool Whether it should have a decline button or not
+     * @todo Implement decline functionality
+     */
+    public $declineButton = false;
+
+    /**
+     * @var string Text on the decline button
+     */
+    public $declinePrivacyButtonText;
+
+    /**
+     * @var string Decline privacy policy button css class
+     */
+    public $declinePrivacyButtonClass = 'btn';
+
+    /**
+     * @var bool Whether the buttons should be forced.
+     */
+    public $forceOutput = false;
+
+    /**
+     * @var string CSS to be applied
+     */
+    public $css = '.privacyPolicyConsent {
+                        position: fixed;
+                        width: 100%;
+                        left: 0;
+                        bottom: 0;
+                        background-color: #A50045;
+                        color: #fff;
+                        z-index: 10000;
+                        display: flex;
+                        padding: 1em;
+                        align-items: center;
+                        justify-content: space-between;
+                    }
+                    .privacyPolicyConsent.top {
+                        top: 0;
+                        bottom: unset;
+                    }
+                    .privacyPolicyConsent a {
+                        color: #fff;
+                    }';
+
+    /**
+     * @var bool Whether to remove css or not
+     */
+    public $removeCSS = false;
+
+    /**
+     * Add a decline cookie. Needed to set the declined privacy policies cookie.
+     */
+    private $addDeclineCookie = false;
+    
+    /**
+     * Translate messages if no widget input
+     */
+    private function setMessages()
+    {
+        if (empty($this->privacyMessage)) {
+            $this->privacyMessage = Module::t('privacy_widget.privacy_message');
+        }
+        if(empty($this->acceptPrivacyButtonText)) {
+            $this->acceptPrivacyButtonText = Module::t('privacy_widget.accept_privacy_button_text');
+        }
+        if(empty($this->declinePrivacyButtonText)) {
+            $this->declinePrivacyButtonText = Module::t('privacy_widget.decline_privacy_button_text');
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        $this->setMessages();
+        Module::onLoad();
+        $privacyPolicy = Yii::$app->request->post('_privacyPolicy', null);
+        if (!empty($privacyPolicy)){
+            if ($privacyPolicy === 'true') {
+                Yii::$app->response->cookies->add(new Cookie(["name" => "_privacyPolicy", "value" => true]));
```

Comment:
Why arent you using the funtions from the trait ... this is why we have the trait - if am right.